samples/Java/src/com/microsoft/azure/servicebus/samples/BasicSendReceiveWithQueueClient.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.servicebus.samples;
+
+import com.microsoft.azure.servicebus.*;
+import com.microsoft.azure.servicebus.primitives.ConnectionStringBuilder;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+
+public class BasicSendReceiveWithQueueClient {
+    // Connection String for the namespace can be obtained from the Azure portal under the
+    // 'Shared Access policies' section.
+    private static final String connectionString = "{connection string}";
+    private static final String queueName = "{queue name}";
+    private static IQueueClient queueClient;
+    private static int totalSend = 100;
+    private static int totalReceived = 0;
+
+    public static void main(String[] args) throws Exception {
+
+        Log.log("Starting BasicSendReceiveWithQueueClient sample");
+
+        // create client
+        Log.log("Create queue client.");
+        queueClient = new QueueClient(new ConnectionStringBuilder(connectionString, queueName), ReceiveMode.PeekLock);
+
+        // send and receive
+        queueClient.registerMessageHandler(new MessageHandler(queueClient), new MessageHandlerOptions(1, false, Duration.ofMinutes(1)));
+        for (int i = 0; i < totalSend; i++) {
+            int j = i;
+            Log.log("Sending message #%d.", j);
+            queueClient.sendAsync(new Message("" + i)).thenRunAsync(() -> { Log.log("Sent message #%d.", j);});
+        }
+
+        while(totalReceived != totalSend) {
+            Thread.sleep(1000);
+        }
+
+        Log.log("Received all messages, exiting the sample.");
+        Log.log("Closing queue client.");
+        queueClient.close();
+    }
+
+    static class MessageHandler implements IMessageHandler {
+        private IQueueClient client;
+
+        public MessageHandler(IQueueClient client) {
+            this.client = client;
+        }
+
+        @Override
+        public CompletableFuture<Void> onMessageAsync(IMessage iMessage) {
+            Log.log("Received message with sq#: %d and lock token: %s.", iMessage.getSequenceNumber(), iMessage.getLockToken());
```

Comment:
>lock token: %s. [](start = 55, length = 15)

better to print message.Body here, so that they can correlate with the SentMessage log and see they indeed got all messages they sent